apps/studio/pages/project/[ref]/logs/explorer/index.tsx

```diff
@@ -130,9 +179,49 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
 
   const handleRun = (value?: string | React.MouseEvent<HTMLButtonElement>) => {
     const query = typeof value === 'string' ? value || editorValue : editorValue
+
     if (value && typeof value === 'string') {
       setEditorValue(value)
     }
+
+    if (sourceType === 'warehouse') {
+      const whQuery = warehouseEditorValue
+
+      if (!warehouseCollections?.length) {
+        toast.error("You don't have any collections in your warehouse yet.")
+        return
+      }
+
+      // Check that a collection name is included in the query
+      const collectionNames = warehouseCollections?.map((collection) => collection.name)
+      const collectionExists = collectionNames?.find((collectionName) =>
+        whQuery.includes(collectionName)
+      )
+
+      if (!collectionExists) {
+        toast.error('Please specify a collection name in the query')
+        return
+      }
+
+      // Check that the user is not trying to query logs tables and warehouse collections at the same time
+      const logsSources = Object.values(LOGS_TABLES)
+      const logsSourceExists = logsSources.find((source) => whQuery.includes(source))
+
+      if (logsSourceExists) {
+        toast.error(
+          'Cannot query logs tables from a warehouse query. Please remove the logs table from the query.'
```

Comment:
Does this imply that the warehouse name cannot match the name of any of the logs sources?
if yes then perhaps we should add some client side validation when creating collections

Reply:
Actually you could, I was thinking about what would happen if users tried to join or query logs tables and wh collections, but wh collections can have the same name so this check is not the best way to do it. I'll remove it for now and see if I come up with something better, this was just for UX.